src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketBuildStatusFactory.java

```diff
@@ -37,6 +37,7 @@ public static BitbucketBuildStatus fromBuild(Run<?, ?> build, BitbucketCICapabil
                 .setDescription(state.getDescriptiveText(build.getDisplayName(), build.getDurationString()));
 
         if (ciCapabilities.supportsRichBuildStatus()) {
+            bbs.setBuildID(key);
```

Comment:
In Jenkins I think we're always going to be using the key (full project name) for the build ID, but they're separate fields in the request so I think it makes sense to treat them separately.